### Importar las librerías

In [1]:
import pandas as pd

### Cargar el archivo chipotle.csv en un dataframe y guardarlo en una variable de nombre chipo

In [2]:
chipo = pd.read_csv('chipotle.csv', delimiter=',')

### Ver las primeras 10 filas

In [3]:
chipo.head(5)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

### ¿Cuál es el número de muestras en el dataset? ¿Y el número de atributos? Mostrar el nombre de los atributos

In [10]:
print(chipo.shape)
print(chipo.columns)
print(' ')
print(chipo.info())

(4622, 5)
Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')
 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
order_id              4622 non-null int64
quantity              4622 non-null int64
item_name             4622 non-null object
choice_description    3376 non-null object
item_price            4622 non-null object
dtypes: int64(2), object(3)
memory usage: 180.6+ KB
None


### ¿Cuántos items se ordenaron en total? Notar que una fila puede contener más de un item en la columna *quantity*

In [13]:
chipo.loc[:, 'quantity'].sum()

4972

### ¿Cuántas órdenes se hicieron en total? Notar que en una orden se pueden comprar muchos productos. Hay una orden por cada *orden_id*

In [14]:
chipo.groupby('order_id').count()

quantity  item_name  choice_description  item_price
order_id                                                     
1                4          4                   2           4
2                1          1                   1           1
3                2          2                   1           2
4                2          2                   2           2
5                2          2                   1           2
6                2          2                   2           2
7                2          2                   1           2
8                2          2                   1           2
9                2          2                   2           2
10               2          2                   1           2
11               2          2                   2           2
12               2          2                   2           2
13               2          2                   1           2
14               2          2                   2           2
15               2          2                   1           2
16               2          2                   1           2
17               2          2                   1           2
18               4          4                   2           4
19               2          2                   1           2
20               4          4                   3           4
21               3          3                   3           3
22               2          2                   1           2
23               2          2                   2           2
24               2          2                   2           2
25               2          2                   1           2
26               2          2                   2           2
27               2          2                   1           2
28               4          4                   3           4
29               2          2                   2           2
30               3          3                   3           3
...            ...        ...                 ...         ...
1805             2          2                   2           2
1806             4          4                   2           4
1807             2          2                   2           2
1808             2          2                   1           2
1809             3          3                   2           3
1810             2          2                   2           2
1811             3          3                   3           3
1812             3          3                   3           3
1813             1          1                   1           1
1814             2          2                   1           2
1815             2          2                   2           2
1816             2          2                   1           2
1817             4          4                   2           4
1818             2          2                   2           2
1819             2          2                   1           2
1820             2          2                   2           2
1821             2          2                   1           2
1822             3          3                   2           3
1823             3          3                   2           3
1824             3          3                   2           3
1825             6          6                   6           6
1826             4          4                   2           4
1827             5          5                   4           5
1828             3          3                   2           3
1829             3          3                   3           3
1830             2          2                   2           2
1831             3          3                   1           3
1832             2          2                   1           2
1833             2          2                   2           2
1834             3          3                   3           3

[1834 rows x 4 columns]

### ¿Cuántos items diferentes se venden?

In [16]:
chipo.groupby('item_name').count().shape

(50, 4)

### Convertir el atributo *item_price* a float. Tener en cuenta que el formato es un string de la forma "$2.39 " por lo que necesitamos remover el primer y el último caracter del string. **IMPORTANTE PARA SEGUIR**

In [41]:
chipo['item_price'].replace([" ", ""], ["$", " "], inplace=True)
print(chipo['item_price'])

0        $2.39 
1        $3.39 
2        $3.39 
3        $2.39 
4       $16.98 
5       $10.98 
6        $1.69 
7       $11.75 
8        $9.25 
9        $9.25 
10       $4.45 
11       $8.75 
12       $8.75 
13      $11.25 
14       $4.45 
15       $2.39 
16       $8.49 
17       $8.49 
18       $2.18 
19       $8.75 
20       $4.45 
21       $8.99 
22       $3.39 
23      $10.98 
24       $3.39 
25       $2.39 
26       $8.49 
27       $8.99 
28       $1.09 
29       $8.49 
         ...   
4592    $11.75 
4593    $11.75 
4594    $11.75 
4595     $8.75 
4596     $4.45 
4597     $1.25 
4598     $1.50 
4599     $8.75 
4600     $4.45 
4601     $1.25 
4602     $9.25 
4603     $9.25 
4604     $8.75 
4605     $4.45 
4606     $1.25 
4607    $11.75 
4608    $11.25 
4609     $1.25 
4610    $11.75 
4611    $11.25 
4612     $9.25 
4613     $2.15 
4614     $1.50 
4615     $8.75 
4616     $4.45 
4617    $11.75 
4618    $11.75 
4619    $11.25 
4620     $8.75 
4621     $8.75 
Name: item_price, Length

### ¿En qué consistió la orden número 1000? ¿Cuánto gastó?

### ¿Cuántas *Veggie Salad Bowl* se compraron?

### ¿Cuántas veces alguien ordenó más de una *Canned Soda*?

### Agregar una columna de nombre *revenue* (ingreso) que contenga el dinero ganado por la compra de los items de la fila

### ¿Cuál es el *revenue* (ingreso) promedio por orden?

### Mostrar las 3 filas que dieron como resultado el mayor *revenue*

### ¿Cuál fue el item más ordenado?

### ¿Qué ingresos produjo el item más ordenado?

### Chipotle decide abrir su primera sucursal en Argentina. Para eso nos pide que realicemos la lista de precios de sus productos

#### Usando nuestro dataframe chipo, agrupemos por *item_name* y calculemos la media de *item_price* para cada item. Creemos un nuevo dataframe de nombre *chipo_precios* que solo contenga los *item_name* como index y una columna con los *item_price*.

#### Chipotle decide reemplazar su producto más ordenado (Chicken Bowl) por algo más local (Chori Bowl). Remover de *chipo_precios* la fila correspondiente y agregar usando concat o append el flamante Chori Bowl que se venderá al módico precio de $5 (dólares). Ordenar *chipo_precios* por index (item_name)

#### Agregar una columna a nuestro dataframe de nombre *item_price_arg* que obtenga el valor en pesos argentinos de cada producto usando la columna *item_price* y el valor del dólar hoy respecto al peso argentino

#### Chipotle ve el valor del dólar en Enero 2018 y decide instalarse en Uruguay. Renombrar la columna anterior a *item_price_uru* usando rename y recalcular sus valores usando el valor del peso uruguayo